### ISSUE TITLE
Modularise cf-view codebase and move non-GUI code to cfplot

### ISSUE THREAD

**Issue by sadielbartholomew** [ajheaps/cf-view#34]
A codebase (re)structuring to move GUI code to self-contained units and anything not related to the GUI, i.e. on the plotting side, out to cf-plot.

Currently all of the code for the package is contained in a single file, [cfview](https://github.com/NCAS-CMS/cf-view/blob/main/cfview) which as a consequence is nearly 10,000 lines long and difficult to follow, manage and navigate. It would be much better to split this into separate files in a modular hierarchy based on functionality and natural groupings of objects, etc.

I will come up with a more concrete plan before starting to move things around, but as an immediate thought it would be good to separate out the GUI view, widget, frame and object classes as well as have a dedicated 'utils' module of helper functions used.

From a skim read of parts of the code I strongly suspect there is some code here which belongs in cf-plot rather than cf-view, so some moving across will likely also be necessary to separate concerns.

I will come up with a plan for what I see is the most appropriate modularisation for cf-plot shortly and post it here as a proposal, before undertaking the work.
